### PR TITLE
Controlling the Motors

### PR DESCRIPTION
**Pull request by stevempope**
This code is completed using the original "blocking" library that comes with the MotorShield, as raised in #117 we need to consider how we approach the ultrasound etc code.

The basic outline is:

Harrys maths with give a number of steps and angle theta. If Theta is negative then we turn the number of steps left, positive means the number of steps right.

Once turned we are on a straight line. Any interrupt from the US sensor (Code pending for that) will result in a calculation for how far along the line we are, and return that result as the current waypoint as per the signature. Normal running will see us reach our destination and we simply return the WP we were handed.

I have been comment heavy for sanity checking but of course this will be cleaned up in future PR's

### CODE REVIEW

**Comment by stevempope** [comment]
Darn, it appears I added the config files by mistake. I will fix and re raise.

**Comment by SamanthaDawsonBanks** [comment]
For the purposes of US interruption assume a global variable (bool) called 'blocked' that is F for good and T on US trigger then just if it every iteration of the loop

**Comment by SamanthaDawsonBanks** [comment]
Also do we need wire lib for this section? Or was that just during dev?
Edit ignore that just saw it in the h file

File: mcomp_C/movement/driving.cpp [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+Adafruit_StepperMotor leftMotor;
+Adafruit_StepperMotor rightMotor;
+
+driving::driving() {
+	//TODO add the Maths parts
+	AFMS = Adafruit_MotorShield(); //The MotorShield
+	leftMotor = AFMS.getStepper(200,1); //360(div)1.8 degree steps, M1 and M2 on the MotorShield
+	rightMotor = AFMS.getStepper(200, 2); //M3 and M4 on the MotorShield
+	leftMotor.setSpeed(30); //Set speed of rotation in RPM, 100Hz * 1.8 Degrees = 1 rotation of wheel every 2 seconds, or 30 RPM.
+	//TODO change to 400Hz, or 120 rpm?
```
Comment (SamanthaDawsonBanks) [inline mcomp_C/movement/driving.cpp:20]: For the faster iteration we would also need to adjust the I2C bus speed

**Review by SamanthaDawsonBanks — APPROVED** [review]

**Review by RyanShoobert — APPROVED** [review]

**Comment by stevempope** [comment]
Thanks for the approval, changes will be actioned for next PR

**Pull request merged**